temporal-sdk/src/test/java/io/temporal/workflow/updateTest/UpdateWithStartTest.java

```diff
@@ -0,0 +1,484 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.workflow.updateTest;
+
+import static org.junit.Assert.*;
+
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import io.temporal.api.common.v1.WorkflowExecution;
+import io.temporal.api.enums.v1.WorkflowIdConflictPolicy;
+import io.temporal.api.enums.v1.WorkflowIdReusePolicy;
+import io.temporal.client.*;
+import io.temporal.testing.internal.SDKTestOptions;
+import io.temporal.testing.internal.SDKTestWorkflowRule;
+import io.temporal.worker.WorkerOptions;
+import io.temporal.workflow.*;
+import io.temporal.workflow.shared.TestMultiArgWorkflowFunctions;
+import io.temporal.workflow.shared.TestWorkflows;
+import java.util.*;
+import java.util.concurrent.ExecutionException;
+import java.util.function.BiFunction;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class UpdateWithStartTest {
+
+  @Rule
+  public SDKTestWorkflowRule testWorkflowRule =
+      SDKTestWorkflowRule.newBuilder()
+          .setWorkerOptions(WorkerOptions.newBuilder().build())
+          .setWorkflowTypes(
+              WorkflowWithUpdateImpl.class,
+              TestUpdatedWorkflowImpl.class,
+              TestMultiArgWorkflowFunctions.TestMultiArgWorkflowImpl.class)
+          .build();
+
+  @Test
+  public void startAndSendUpdateTogether() throws ExecutionException, InterruptedException {
+    WorkflowClient workflowClient = testWorkflowRule.getWorkflowClient();
+
+    WorkflowOptions options = createOptions();
+    TestWorkflows.WorkflowWithUpdate workflow =
+        workflowClient.newWorkflowStub(TestWorkflows.WorkflowWithUpdate.class, options);
+
+    UpdateWithStartWorkflowOperation<String> updateOp =
+        UpdateWithStartWorkflowOperation.newBuilder(workflow::update, 1, "Hello Update")
+            .setWaitForStage(WorkflowUpdateStage.COMPLETED)
+            .build();
+
+    WorkflowUpdateHandle<String> handle1 =
+        WorkflowClient.updateWithStart(workflow::execute, updateOp);
```

Comment:
Can you show how to get/wait on the workflow execution/result?

Reply:
Line 77 has
```
assertEquals("Hello Update complete", workflow.execute());
```
isn't that how you do that in Java? (ie call the WorkflowMethod again)

Reply:
Does that start a different workflow? Update with start I would expect to be a single call that does both update and start at the same time and I would expect to be able to get the execution (or result) of the workflow alongside the handle of the update. Would not expect to make a separate workflow call potentially starting it again.

Reply:
> Would not expect to invoke the method I already passed to update with start.

Looking at BatchRequest, I see that the tests are doing `WorkflowStub.fromTyped(workflow).getResult(<type>)`. I've updated this test accordingly: https://github.com/temporalio/sdk-java/pull/2199/commits/db5956e539273cd7a7cc9535d4c560019c8547b9

I'll say that having to pass in the workflow result type again is not ideal. But right now I fail to see an obvious alternative without that problem. 

> And what if I just want the workflow execution?

That's done via `handle.getExecution()`; for example see
```
assertEquals(options.getWorkflowId(), handle1.getExecution().getWorkflowId());
```

Reply:
I think getting started workflow information from the update could be a bit confusing for users, but I think we all now recognize Java is just going to be different from everyone else.